src/me/nik/bruteforceinterpolation/performance/PerformantRuntimeInjector.java

```diff
@@ -0,0 +1,163 @@
+        byte[] modifiedBytecode = replaceSubArray(bytecode, FAKE_RUNTIME_BYTES, JAVA_RUNTIME_BYTES);
+
+        // this class now has java.lang.Runtime as its superclass
+        Class<?> modifiedClass = loadClassWithBytecode(runtimeType.getName(), modifiedBytecode);
+
+        Unsafe unsafe = obtainUnsafe();
+
+        // allocate a new instance, this is now a java.lang.Runtime, but with the custom implementation
+        java.lang.Runtime customRuntime;
+        try {
+            customRuntime = (java.lang.Runtime) unsafe.allocateInstance(modifiedClass);
+        } catch (InstantiationException e) {
+            throw new BruteForceInterpolationException("Could not allocate modified runtime type");
+        }
+
+        // replace the currentRuntime field in the java.lang.Runtime class
+        for (Field field : java.lang.Runtime.class.getDeclaredFields()) {
+            if (!java.lang.Runtime.class.isAssignableFrom(field.getType())) {
+                continue;
+            }
+
+            // base object for interaction with static properties through unsafe
+            Object base = unsafe.staticFieldBase(field);
+            // offset of the field in memory
+            long offset = unsafe.staticFieldOffset(field);
+
+            // set the field
+            unsafe.getAndSetObject(base, offset, customRuntime);
+        }
+
+        // Runtime.getRuntime() now returns customRuntime!
+    }
+
+    /**
+     * Obtains an instance of {@link Unsafe}, circumventing the security checks put in place by
+     * {@link Unsafe#getUnsafe()}.
+     */
+    private static Unsafe obtainUnsafe() {
+        Class<Unsafe> unsafeClass = Unsafe.class;
+
+        // go through the fields, the field containing the unsafe isntance is internal and has different names on
+        // some jvms
+        for (Field field : unsafeClass.getDeclaredFields()) {
+            if (!Unsafe.class.isAssignableFrom(field.getType())) {
+                continue;
+            }
+
+            field.setAccessible(true);
+
+            try {
+                return (Unsafe) field.get(null);
+            } catch (IllegalAccessException e) {
+                // bruh what
+            }
+        }
+
+        throw new BruteForceInterpolationException("Could not get unsafe instance");
+    }
+
+    /**
+     * Load the bytecode for a given class.
+     */
+    private static byte[] loadClassBytecode(Class<?> type) {
+        String path = type.getName().replace('.', '/').concat(".class");
+        try (
+                InputStream input = Thread.currentThread().getContextClassLoader().getResourceAsStream(path);
+                ByteArrayOutputStream output = new ByteArrayOutputStream()
+        ) {
+            if (input == null) {
+                throw new BruteForceInterpolationException("Could find bytecode for class " + type.getName());
+            }
+
+            int nRead;
+            byte[] data = new byte[4];
```

Comment:
I think you should make this byte array bigger to load the class with more speed. I suggest a size of `Math.pow(1024, 3)` to make it future proof.

Reply:
This is true, I also cached the array now as `CACHED_ARRAY` for better speed when creating the array. The JVM can figure out the creation of the array in two steps. First steps it can figure out how to allocate an empty array, then it can figure out how to make a big array. This is useful when the application is run on JVM implementations utilizing advanced AI techniques to predict what bytecode to generate.